app/assets/javascripts/application.js

```diff
@@ -238,6 +238,19 @@ var Downloads = {
     Downloads.observeGUIOSFilter();
     Downloads.observePopState();
     Downloads.filterGUIS();
+    Downloads.autoDownload();
+  },
+
+  autoDownload: function() {
+    var ref = document.referrer;
+    var src = $('#auto-download-link').attr('href');
+
+    if (ref && src && ref.indexOf(document.location.origin) === 0) {
+      $('.download .hide').removeClass('hide');
+      $('<iframe frameborder="0" height="1" width="1"></iframe>')
```

Comment:
this seems weird. Not an js expert, but there's probably a better way to achieve what you are trying to do in a cleaner way. What are you achieving with this? Will try to see if there's another solution

Reply:
You're right.  I just took the `iframe` markup and included that in the method.  Setting the location is a cleaner way to achieve triggering the download.